packages/ckeditor5-table/src/tableclipboard.ts

```diff
@@ -153,54 +153,157 @@ export default class TableClipboard extends Plugin {
+			endColumn: Math.min( selectionWidth, pastedDimensions.width ) - 1
+		};
+
+		pastedTable = cropTableToDimensions( pastedTable, cropDimensions, writer );
+
+		// Content table to which we insert a pasted table.
+		const selectedTable = selectedTableCells[ 0 ].findAncestor( 'table' )!;
+
+		const cellsToSelect = this._replaceSelectedCellsWithPasted( pastedTable, pastedDimensions, selectedTable, selection, writer );
+
+		if ( this.editor.plugins.get( 'TableSelection' ).isEnabled ) {
+			// Selection ranges must be sorted because the first and last selection ranges are considered
+			// as anchor/focus cell ranges for multi-cell selection.
+			const selectionRanges = tableUtils.sortRanges( cellsToSelect.map( cell => writer.createRangeOn( cell ) ) );
+
+			writer.setSelection( selectionRanges );
+		} else {
+			// Set selection inside first cell if multi-cell selection is disabled.
+			writer.setSelection( cellsToSelect[ 0 ], 0 );
+		}
+
+		return cellsToSelect;
+	}
+
+	/**
+	 * Inserts a marker map into the tree in the form of fake `$marker` elements (at the beginning of the marker range and at the end).
+	 * The fake elements will be used in further steps to restore the markers after table modifications.
+	 */
+	private _convertMarkersToElements( markers: Map<string, Range>, writer: Writer ) {
+		const sortedMarkers = [ ...markers.entries() ]
+			.flatMap( ( [ name, { start, end } ] ) => [
+				{ position: start, name },
+				{ position: end, name }
+			] )
+			// Markers position is sorted backwards to ensure that the insertion of fake markers will not change
+			// the position of the next markers.
+			.sort( ( { position: posA }, { position: posB } ) => posA.isBefore( posB ) ? 1 : -1 );
+
+		for ( const { position, name } of sortedMarkers ) {
+			const fakeMarker = writer.createElement( '$marker', { 'data-name': name } );
+
+			writer.insert( fakeMarker, position );
+		}
+	}
+
+	/**
+	 * Restores the markers to the tree after the table modifications are finished.
+	 * It inserts them in place of each fake element (which are then removed).
+	 */
+	private _convertElementsToMarkers( replacedCells: Array<Element>, writer: Writer ) {
+		const markersFakeElements = replacedCells
+			.flatMap( cell => Array.from( writer.createRangeOn( cell ) ) )
+			.reduce<Record<string, Array<Item>>>(
+				( acc, { item: cellItem } ) => {
+					const markerAttribute = cellItem.getAttribute( 'data-name' ) as string | undefined;
+
+					if ( !markerAttribute ) {
+						return acc;
+					}
+
+					if ( !acc[ markerAttribute ] ) {
+						acc[ markerAttribute ] = [];
+					}
+
+					acc[ markerAttribute ].push( cellItem );
+					return acc;
+				},
+				{}
+			);
+
+		// sometimes fake markers are duplicated and there is more than two elements in array.
+		// it can happen if you copy column from first table with single marker and then paste
+		// it as row to second table
+		for ( const [ markerName, positions ] of Object.entries( markersFakeElements ) ) {
```

Comment:
imo it's not possible due to complex table crop logic. Basically we don't know where pasted cells are after pasting. For example - if you copy whole column to row it will duplicate first cell X times somewhere in the table. Current solution is good enough and probably will be merged together with solution from clipboard.